GPflowOpt/acquisition.py

```diff
@@ -456,3 +458,42 @@ def __mul__(self, other):
             return AcquisitionProduct(self.operands.sorted_params + other.operands.sorted_params)
         else:
             return AcquisitionProduct(self.operands.sorted_params + [other])
+
+
+class MCMCAcquistion(AcquisitionSum):
+    def __init__(self, acquisition, n_slices, **kwargs):
+        assert isinstance(acquisition, Acquisition)
+        assert n_slices > 0
+
+        copies = [copy.deepcopy(acquisition) for _ in range(n_slices - 1)]
+        for c in copies:
+            c._optimize_restarts = 0
+
+        super(MCMCAcquistion, self).__init__([acquisition] + copies)
+        self._sample_opt = kwargs
+        self._update_hyper_draws()
```

Comment:
If not used with `BayesianOptimizer` and models hold data, set_data might not be called. In generally, you'd expect the object to be functional after the constructor.